src/test/groovy/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTriggerSpec.groovy

```diff
@@ -0,0 +1,42 @@
+package pl.tomaszdziurko.jvm_bloggers
+
+import pl.tomaszdziurko.jvm_bloggers.blog_posts.BlogPostsFetchingScheduler;
+import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPostRepository
+import pl.tomaszdziurko.jvm_bloggers.blogs.BloggersDataFetchingScheduler;
+import pl.tomaszdziurko.jvm_bloggers.blogs.domain.BlogRepository;
+import spock.lang.Specification;
+import spock.lang.Subject
+
+class InitialBlogDataPopulationTriggerSpec extends Specification {
+
+    def BlogRepository blogRepository = Mock()
+    def BlogPostRepository blogPostRepository = Mock()
+    def BloggersDataFetchingScheduler bloggersDataFetchingScheduler = Mock();
+    def BlogPostsFetchingScheduler blogPostsFetchingScheduler = Mock();
+
+    @Subject
+    def InitialBlogDataPopulationTrigger tested = new InitialBlogDataPopulationTrigger(blogRepository, blogPostRepository, bloggersDataFetchingScheduler, blogPostsFetchingScheduler)
+
+    def "Should trigger bloggers data population if there is no data"() {
+        given:
+            1 * blogRepository.count() >> 0
```

Comment:
It's a verification if the method was actually invoked (like 'verify(blogrepository).count()' in Mockito). It ensures the test will fail if the method call was removed from 'InitialBlogDataPopulationTrigger'.